apps/web/app/vagas/page.tsx

```diff
@@ -0,0 +1,203 @@
+'use client'
+
+import { useState } from 'react'
+import { openings20230809 as Openings } from '../../../manual-email-sender/src/openings-email/2023-08-09/openings'
+import { Input } from 'app/components/ui/input'
+import { Button } from 'app/components/ui/button'
+import JobArticle, { JobArticleProps } from 'app/components/ui/JobArticle'
+import { ToogleGroup, ToogleGroupItem } from 'app/components/ui/ToggleGroup'
+import {
+  Dialog,
+  DialogTrigger,
+  DialogContent,
+  DialogHeader,
+  DialogFooter,
+  DialogTitle,
+  DialogDescription,
+} from 'app/components/ui/dialog'
+import { Search } from 'lucide-react'
+
+export default function Page() {
+  const [searchText, setSearchText] = useState<string>('')
+  const [selectedLocations, setSelectedLocations] = useState<string[]>([])
+  const [selectedLanguages, setSelectedLanguages] = useState<string[]>([])
+  const [selectedSkills, setSelectedSkills] = useState<string[]>([])
+
+  const { localOpenings, globalOpenings } = Openings
+
+  const jobList = localOpenings.concat(globalOpenings) as JobArticleProps[]
+
+  const availableLocations = [
+    ...new Set(jobList.map((job) => job.location).flat()),
+  ]
+  const availableLanguages = [
+    ...new Set(jobList.map((job) => job.language).flat()),
+  ]
+  const availableSkills = [...new Set(jobList.map((job) => job.skills).flat())]
```

Comment:
What do you think about using only one iteration in `jobList` using Recuder. 
> I didn't make a test it's just an idea to try to improve these three loops

```suggestion
const available = jobList.reducer((obj, currJob) => {
  obj.locations.add(currJob.location)
  obj.languages.add(currJob.language)
  obj.skills.add(currJob.skills)
}, {
  locations: new Set(),
  languages: new Set(),
  skills: new Set()
})

// you can use like:
Array.from(available.locations) // ['location'...]
```

Reply:
Sorry I made a mistake, it's `reduce` not `reducer` 😅

Reply:
Sorry but I didn't understand how it would work with reducer. Could you please explain? I've tested the code and it didn't work. By the way, I made a few changes. I will commit right away.

Reply:
The use of reduce was only an option to avoid making three iterations. The way that you did implement it's great 🚀